File: ObjectProgrammingJava/projekt po - 2/src/projekt/zwierzeta/Lis.java
package projekt.zwierzeta;

import projekt.Organizm;
import projekt.Wypisanie;
import projekt.obsluga.Para;
import projekt.swiat.Swiat;

import java.awt.*;
import java.util.Random;
import java.util.Vector;

public class Lis extends Zwierze implements Wypisanie {
    public Lis(Swiat swiat, int x, int y, boolean czyZyje){
        polozenieX = x;
        polozenieY = y;
        sila = 3;
        inicjatywa = 7;
        kolor = new Color(150, 58, 0);
        oznaczenie = 'L';
        this.swiat = swiat;
        this.czyZyje = czyZyje;
        this.czyMozeRozmnazac = 0;
        wiek = 0;
    }

    public void akcja(){
        przygotujOrganizm();
        String status = "";
        Vector<String> kom = swiat.getKomentarz();
        Organizm[][] swiatOrganizmow = swiat.getSwiatOrganizmow();
        Vector<Para> pola = szukajPola(polozenieX, polozenieY);
        if (pola.size() > 0) {
            Random random = new Random();
            int losuj = random.nextInt(pola.size());
            if (swiatOrganizmow[pola.get(losuj).getL()][pola.get(losuj).getR()] == null) {
                swiatOrganizmow[polozenieX][polozenieY] = null;
                status = Wypisz() + " PORUSZYL SIE NA POZYCJE ";
                polozenieX = pola.get(losuj).getL();
                polozenieY = pola.get(losuj).getR();
                status +=  String.valueOf(polozenieX)+ ":" + String.valueOf(polozenieY);
                swiatOrganizmow[polozenieX][polozenieY] = this;
            }
            else if(swiatOrganizmow[pola.get(losuj).getL()][pola.get(losuj).getR()] != null){
                kolizja(swiatOrganizmow[pola.get(losuj).getL()][pola.get(losuj).getR()]);
            }
        }
        else {
            status = Wypisz() + " NIE PORUSZYL SIE Z POWODU BRAKU MIEJSCA";
        }
        kom.add(status);
    }

    @Override
    public String Wypisz() {
        return "Lis" + getKoncowka();
    }

    Vector<Para> szukajPola(int x, int y) {
        Organizm[][] swiatOrganizmow = swiat.getSwiatOrganizmow();
        Vector<Para> vect =  new Vector<>();
        if (x - 1 > 0 && y - 1 > 0 && (swiatOrganizmow[x - 1][y - 1] == null || swiatOrganizmow[x - 1][y - 1].getSila()<sila)) {
            vect.add(new Para(x - 1, y - 1));
        }
        if (x - 1 > 0 && (swiatOrganizmow[x - 1][y] == null || swiatOrganizmow[x - 1][y].getSila() < sila)) {
            vect.add(new Para(x - 1, y));
        }
        if (x - 1 > 0 && y + 1 < swiat.getSizeY() && (swiatOrganizmow[x - 1][y + 1] == null || swiatOrganizmow[x - 1][y + 1].getSila() < sila)) {
            vect.add(new Para(x - 1, y + 1));
        }
        if (y - 1 > 0 && (swiatOrganizmow[x][y - 1] == null || swiatOrganizmow[x][y - 1].getSila() < sila)) {
            vect.add(new Para(x, y - 1));
        }
        if (y + 1 < swiat.getSizeY() && (swiatOrganizmow[x][y + 1] == null || swiatOrganizmow[x][y + 1].getSila() < sila)) {
            vect.add(new Para(x, y + 1));
        }
        if (x + 1 < swiat.getSizeX() && y - 1 > 0 && (swiatOrganizmow[x + 1][y - 1] == null || swiatOrganizmow[x + 1][y - 1].getSila() < sila)) {
            vect.add(new Para(x + 1, y - 1));
        }
        if (x + 1 < swiat.getSizeX() && (swiatOrganizmow[x + 1][y] == null || swiatOrganizmow[x + 1][y].getSila() < sila)) {
            vect.add(new Para(x + 1, y));
        }
        if (x + 1 < swiat.getSizeX() && y + 1 < swiat.getSizeY() && (swiatOrganizmow[x + 1][y + 1] == null || swiatOrganizmow[x + 1][y + 1].getSila() < sila)) {
            vect.add(new Para(x + 1, y + 1));
        }
        return vect;
    }
}

File: ObjectProgrammingPython/Zwierzeta/lis.py
import random

from Obsluga.para import Para
from Zwierzeta.zwierze import Zwierze


class Lis(Zwierze):
    def __init__(self, Swiat, x, y, czyZyje):
        super(Lis, self).__init__(Swiat, x, y, czyZyje)
        self.sila = 3
        self.inicjatywa = 7
        self.kolor = "orange"
        self.oznaczenie = "L"
        self.czyMozeRozmnazac = 0
        self.wiek = 0

    def akcja(self):
        self.przygotujOrganizm()
        status = ""
        kom = self.swiat.getKomentarz()
        swiatOrganizmow = self.swiat.getSwiatOrganizmow()
        pola = self.szukajPola(self.polozenieX, self.polozenieY)
        if len(pola):
            losuj = random.randint(0,len(pola)-1)
            if swiatOrganizmow[pola[losuj].getX()][pola[losuj].getY()] is None:
                swiatOrganizmow[self.polozenieX][self.polozenieY] = None
                status = self.Wypisz() + " PORUSZYL SIE NA POZYCJE "
                self.polozenieX = pola[losuj].getX()
                self.polozenieY = pola[losuj].getY()
                status += str(self.polozenieX) + ":" + str(self.polozenieY)
                swiatOrganizmow[self.polozenieX][self.polozenieY] = self
            else:
                self.kolizja(swiatOrganizmow[pola[losuj].getX()][pola[losuj].getY()])
        else:
            status = self.Wypisz() + " NIE PORUSZYL SIE Z POWODU BRAKU MIEJSCA"
        kom.append(status)

    def szukajPola(self, x, y):
        swiatOrganizmow = self.swiat.getSwiatOrganizmow()
        sizeX = self.swiat.getSizeX()
        sizeY = self.swiat.getSizeY()
        listaPustych = []
        if x - 1 >= 0 and y - 1 >= 0 and (swiatOrganizmow[x - 1][y - 1] is None or swiatOrganizmow[x - 1][y - 1].getSila()<self.sila):
            listaPustych.append(Para(x - 1, y - 1))
        if x - 1 >= 0 and (swiatOrganizmow[x - 1][y] is None or swiatOrganizmow[x - 1][y].getSila()<self.sila):
            listaPustych.append(Para(x - 1, y))
        if x - 1 >= 0 and y + 1 < sizeY and (swiatOrganizmow[x - 1][y + 1] is None or swiatOrganizmow[x - 1][y + 1].getSila()<self.sila):
            listaPustych.append(Para(x - 1, y + 1))
        if y - 1 >= 0 and (swiatOrganizmow[x][y - 1] is None or swiatOrganizmow[x][y - 1].getSila()<self.sila):
            listaPustych.append(Para(x, y - 1))
        if y + 1 < sizeY and (swiatOrganizmow[x][y + 1] is None or swiatOrganizmow[x][y + 1].getSila()<self.sila):
            listaPustych.append(Para(x, y + 1))
        if x + 1 < sizeX and y - 1 >= 0 and (swiatOrganizmow[x + 1][y - 1] is None or swiatOrganizmow[x + 1][y - 1].getSila()<self.sila):
            listaPustych.append(Para(x + 1, y - 1))
        if x + 1 < sizeX and (swiatOrganizmow[x + 1][y] is None or swiatOrganizmow[x + 1][y].getSila()<self.sila):
            listaPustych.append(Para(x + 1, y))
        if x + 1 < sizeX and y + 1 < sizeY and (swiatOrganizmow[x + 1][y + 1] is None or swiatOrganizmow[x + 1][y + 1].getSila()<self.sila):
            listaPustych.append(Para(x + 1, y + 1))
        return listaPustych

    def Wypisz(self):
        return "Lis" + self.koncowka()



